### PR TITLE
[opentitantool] Refactor cw310 json files

### PR DESCRIPTION
**Pull request by jesultra**
Make use of "invert" setting to consolidate json files such that opentitan.json is shared, and opentitan_cw310.json / opentitan_cw340.json declare the differences between setups.

Remove hard-coded list of signal direction and default levels from Rust code, and make use of json default levels.  This means that the directions/levels are not applied each time the driver is initialized, but instead only when "transport init" is invoked.  So "transport init" should be done before operations, when the state of the debugger is not known for sure, such as before each test run.

### CODE REVIEW

**Review by engdoreis — APPROVED** [review]

File: rules/opentitan_test.bzl [hunk 1]
```diff
@@ -236,6 +236,7 @@ def fpga_params(
         tags = _BASE_PARAMS["tags"],
         test_runner = _BASE_PARAMS["test_runner"],
         test_cmds = [
+            "--exec=\"transport init\"",
```
Comment (pamaury) [inline rules/opentitan_test.bzl:239]: You probably need to update the [new rules as well](https://github.com/lowRISC/opentitan/blob/master/hw/top_earlgrey/BUILD#L55).

**Comment by pamaury** [comment]
I have a question: what happens if someone forgets to run `transport init`? Maybe this is a naming problem but it feels like failing to init should be an error, or that it should automatically init on first use. Or maybe `init` would need to be renamed to something more explicit if we only use it for pins. Also you change only adds it for fpga targets and not verilator or dv that also use a transport so it might be confusing?

**Comment by jesultra** [comment]
> I have a question: what happens if someone forgets to run `transport init`?

Failing to run `transport init` would mean that you run with whatever state that the previous user left the transport (debugger) in.

"State" in this case refers to the input/output setting of GPIO pins, and their initial level and/or weak pull direction, as well as pinmux-like settings for declaring which pins have UART, SPI, I2C, or other special functions. In general, the declarations used for such initialization will be a combination of declarations relating to the physical wiring connections between the debugger and the Titan chip, as well as declarations relating to the firmware to run on the Titan chip.

It would be nice to automatically init on first use, or detecting lack of initialization.  But subsequent tests could be using separate configuration files if they use separate firmware with different purpose, and it is not possible for `opentitantool` to know when one test is over and the next begins.

> Also you change only adds it for fpga targets and not verilator or dv that also use a transport so it might be confusing?
Some of the other transports are less flexible, so one could get away with not re-initializing.  I think we should, though, but I am not familiar enough with the bazel files to know all the places.  I will try revising.

File: rules/opentitan_test.bzl (hunk 1 above)
Comment (jesultra) [reply rules/opentitan_test.bzl:239]: I seems that Douglas beat me to it.

**Pull request merged**